### PR TITLE
Fix #16173: SJS: Fix the detection of inferred types of `= js.native`.

### LINKED ISSUE #16173: Scala.js: Literal types not accepted for JS native methods/values
## Compiler version

3.2.0

## Minimized code

```Scala
import scala.scalajs.js

@js.annotation.JSGlobal("a")
@js.native
def string: String = js.native // ok

@js.annotation.JSGlobal("a")
@js.native
def a: "a" = js.native // error: The type of a must be explicitly specified because it is JS native.
// same with `var`/`val`
```

## Output

```
The type of a must be explicitly specified because it is JS native.
```

## Expectation
The type is specified, so it should be accepted.

I see this in the implementation of the check:
```scala
      if (tree.tpt.span.isSynthetic)
        report.error(i"The type of ${tree.name} must be explicitly specified because it is JS native.", tree)
```

Perhaps it is the case that the literal type tree is considered synthetic somehow?

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
That is indeed a better fix, thanks. I've updated the PR.

That said, is it still not wrong that literal type trees get a synthetic span? 🤔

**Comment by smarter** [comment]
I guess you could do `override def span = ref.span` in SingletonTypeTree since it really represents the same thing as the tree it wraps?

File: compiler/src/dotty/tools/dotc/transform/sjs/PrepJSInterop.scala [hunk 1]
```diff
@@ -981,7 +981,7 @@ class PrepJSInterop extends MacroTransform with IdentityDenotTransformer { thisP
 
       // Check that the resul type was explicitly specified
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/transform/sjs/PrepJSInterop.scala:982]: drive-by typo suggestion:
```suggestion
      // Check that the result type was explicitly specified
```

Comment (sjrd) [reply compiler/src/dotty/tools/dotc/transform/sjs/PrepJSInterop.scala]: Thanks, fixed.

**Review by smarter — APPROVED** [review]

**Pull request merged**